tests/routes/datasets/test_datasets.py

```diff
@@ -70,16 +73,54 @@ async def test_datasets(async_client):
     assert len(rows) == 1
     assert rows[0][0] == "mr_manager123"
 
-    new_payload = {"metadata": {"title": "New Title"}}
-    response = await async_client.patch(f"/dataset/{dataset}", json=new_payload)
-    assert response.status_code == 200
-    assert response.json()["data"]["metadata"] != payload["metadata"]
-    assert response.json()["data"]["metadata"]["title"] == "New Title"
-    assert (
-        response.json()["data"]["metadata"]["data_language"]
-        == payload["metadata"]["data_language"]
+    with patch("app.routes.datasets.dataset.get_rw_user", return_value=NEW_OWNER):
+        new_payload = {
+            "metadata": {"title": "New Title"},
+            "owner_id": "new_owner_id123",
+        }
+        response = await async_client.patch(f"/dataset/{dataset}", json=new_payload)
+        assert response.status_code == 200
+        assert response.json()["data"]["metadata"] != payload["metadata"]
+        assert response.json()["data"]["metadata"]["title"] == "New Title"
+        assert (
+            response.json()["data"]["metadata"]["data_language"]
+            == payload["metadata"]["data_language"]
+        )
+
+        async with ContextEngine("READ"):
+            rows = await db.all(
+                f"SELECT owner_id FROM datasets WHERE dataset = '{dataset}';"
+            )
+
+        assert len(rows) == 1
+        assert rows[0][0] == "new_owner_id123"
+
+        # assign original owner back
+        from app.main import app
+
+        app.dependency_overrides[get_manager] = get_new_owner_mocked
+        new_payload = {"owner_id": "mr_manager123"}
```

Comment:
To be fully correct, I would think that you need to put this code from line 101 to 104 inside a different 'with patch' that forces get_rw_user to return MANAGER (i.e. the User object that get_manager_mocked is returning), so it matches up with the owner_id "mr_manager123" that you are going to be looking up.  The test works now because NEW_OWNER is also a manager, which is what you require.

Feel free to ignore this for now.